File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__get_upsidedown.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int[] dx = { 0, 0, -1, 0, 1 };
        int[] dy = { -1, 1, 0, 0, 0 };
        int[] hw = ReadLine().Split().Select( int.Parse ).ToArray();
        string[] s = new string[hw[0]];

        foreach( int i in Range( 0, hw[0] ) ) {
            s[i] = ReadLine();
        }

        int[] yx = ReadLine().Split().Select( int.Parse ).ToArray();

        foreach( int i in Range( 0, 5 ) ) {
            int cx = yx[1] + dx[i];
            int cy = yx[0] + dy[i];

            if( cx < 0 || hw[1] <= cx || cy < 0 || hw[0] <= cy ) {
                continue;
            }

            char[] c = s[cy].ToCharArray();

            c[cx] = c[cx] == '.' ? '#' : '.';
            s[cy] = string.Join( "", c );
        }
        WriteLine( string.Join( System.Environment.NewLine, s ) );
        return;
    }
}

File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__get_upsidedown.py
# 実行時間 10ms
h, w = list(map(int, input().split()))
a = [[False for i in range(w)] for j in range(h)]
DY = [-1, 1, 0, 0, 0]
DX = [0, 0, -1, 0, 1]

for i in range(h):
    s = input()

    for j, c in enumerate(s):
        if c == "#":
            a[i][j] = True

y, x = list(map(int, input().split()))

for i in range(5):
    o = y + DY[i]
    p = x + DX[i]

    if 0 <= o < h and 0 <= p < w:
        a[o][p] = not a[o][p]
for i in range(h):
    for j in range(w):
        print("#" if a[i][j] else ".", end="")
    print()
